libraries/AP_Compass/AP_Compass_IST8310.cpp

```diff
@@ -0,0 +1,176 @@
+/// -*- tab-width: 4; Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil -*-
+/*
+ * Copyright (C) 2016  Emlid Ltd. All rights reserved.
+ *
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * Driver by Georgii Staroselskii, Sep 2016
+ */
+
+#include "AP_Compass_IST8310.h"
+
+#include <AP_HAL/AP_HAL.h>
+
+#include <utility>
+
+#include <AP_HAL/utility/sparse-endian.h>
+#include <AP_Math/AP_Math.h>
+
+#define WAI_REG 0x0
+#define DEVICE_ID 0x10
+
+#define CNTL1_REG 0xA
+#define SINGLE_MEASUREMENT_MODE 0x1
+#define ODR_100HZ 0x6
+
+#define STAT1_REG 0x2
+#define DATA_RDY 0x1
+
+extern const AP_HAL::HAL &hal;
+
+AP_Compass_Backend *AP_Compass_IST8310::probe(Compass &compass,
+                                             AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev)
```

Comment:
already add support for rotation since this will be needed later. See AP_Compass_LIS3MDL or AP_Compass_LSM9DS1... it needs an enum Rotation as argument.